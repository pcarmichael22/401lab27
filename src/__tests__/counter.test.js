import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Counter from '../components/counter/counter';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing our Counter component', () => {
  it('Counter changes state up on Click +', () => {
    let app = mount(<Counter />);
    let button = app.find('.up');
    expect(button).toBeDefined();
    expect(app.state('count')).toBe(0);
    button.simulate('click');
    expect(app.state('count')).toBe(1);
  });

  it('Counter changes state down on Click -', () => {
    let app = mount(<Counter />);
    let button = app.find('.down');
    expect(button).toBeDefined();
    expect(app.state('count')).toBe(0);
    button.simulate('click');
    expect(app.state('count')).toBe(-1);
  });

  it('Assert that state is being transferred to the DOM', () => {
      let app = mount(<Counter />);
      let span = app.find('.count');
      expect(span).toBeDefined();
  })

  it('Renders correctly', () => {
    const renderTree = renderer.create(<Counter />).toJSON();
    expect(renderTree).toMatchSnapshot();
  })
});