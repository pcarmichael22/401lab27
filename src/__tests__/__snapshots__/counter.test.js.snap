// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing our Counter component Renders correctly 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;

exports[`Testing our Counter component render correctly 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
